src/bin.ts

```diff
@@ -54,16 +54,43 @@ v8flags(function (err, v8flags) {
   const proc = spawn(
     process.execPath,
     nodeArgs.concat(join(__dirname, '_bin.js'), scriptArgs),
-    { stdio: 'inherit' }
+    {
+      // We need to run in detached mode so to avoid
+      // automatic propagation of signals to the child process.
+      // This is necessary because by default, keyboard interrupts
+      // are propagated to the process tree, but `kill` is not.
+      //
+      // See: https://nodejs.org/api/child_process.html#child_process_options_detached
+      detached: true,
+
+      // Pipe all input and output to this process
+      stdio: 'inherit'
+    }
   )
 
-  proc.on('exit', function (code: number, signal: string) {
-    process.on('exit', function () {
-      if (signal) {
-        process.kill(process.pid, signal)
-      } else {
-        process.exit(code)
-      }
-    })
+  // Ignore signals, and instead forward them to the
+  // child process
+  const forward = (signal: string) => process.on(signal, () => proc.kill(signal))
```

Comment:
Probably easier just to make this an array loop. The comments are super verbose and mostly unnecessary too. E.g.

```js
['SIGINT', 'SIGTERM', 'SIGWINCH'].forEach(signal => process.on(signal, () => proc.kill(signal))
```

Reply:
TBH I never remember what `SIGWINCH` does, hence why I wrote it in a slightly more verbose way. If you consider this a blocker, I will change it, of course.